### PR TITLE
[WIP] Enable obu

### PR DESCRIPTION
**Pull request by ycho**
#246

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -65,14 +65,111 @@ impl Frame {
...
 }
 
 impl Sequence {
-    pub fn new() -> Sequence {
+    pub fn new(width: usize, height: usize) -> Sequence {
+        let mut operating_point_idc = [0 as u16; MAX_NUM_OPERATING_POINTS];
+        let mut level = [[1, 2 as usize]; MAX_NUM_OPERATING_POINTS];
+        let mut tier = [0 as usize; MAX_NUM_OPERATING_POINTS];
+
+        for i in 0..MAX_NUM_OPERATING_POINTS {
```
Comment (lu-zero) [inline src/lib.rs]: Why setting them here again if you already set them just right above?

File: src/lib.rs [hunk 2]
```diff
@@ -366,6 +474,57 @@ impl<'a> UncompressedHeader for BitWriter<'a, BE> {
...
+    fn write_sequence_header_obu(&mut self, seq: &Sequence, fi: &FrameInvariants)
+        -> Result<(usize), std::io::Error> {
+        let mut size: usize = 0;
+
+        self.write(3, seq.profile)?; // profile 0, 3 bits
+        self.write(1, 0)?; // still_picture
+        self.write(1, 0)?; // reduced_still_picture
+
+        if false {	// if reduced_still_picture == true
+            assert!(false);
```
Comment (lu-zero) [inline src/lib.rs]: what about 
``` rust
panic!("some informative message")
```
or using 
``` rust 
unimplemented!("some informative message")
```

File: src/lib.rs (hunk 1 above)
Comment (ycho) [reply src/lib.rs]: Yeap- right. Some members of Sequence are set when it is constructed and others are set right before it is written to bitstream, i.e. after frame is finished for encoding.
Then, I haven't really decided where to init them, so duplicated. Thanks for catching them.

File: src/lib.rs [hunk 3]
```diff
@@ -758,45 +758,75 @@ fn aom_uleb_size_in_bytes(mut value: u64) -> usize {
...
-        let obu_payload_size_uleb = aom_uleb_size_in_bytes(obu_payload_size);
-
-        /*const size_t length_field_size =
-            obu_memmove(obu_header_size, obu_payload_size, data);
-        if (write_uleb_obu_size(obu_header_size, obu_payload_size, data) !=
-            AOM_CODEC_OK) {
-        			return AOM_CODEC_ERROR;
-        }*/
+        packet.write(&buf2).unwrap();
+        buf2.clear();
```
Comment (ycho) [inline src/lib.rs]: @lu-zero : I'd like this checked by you. So, if I call 
packet.write(&buf1).unwrap(); and packet.write(&buf2).unwrap(); in order,
bu1 and bu2 is written to packet buffer in order, correct?

Comment (lu-zero) [reply src/lib.rs]: I think so.

**Comment by ycho** [comment]
Fix already submitted by mbx.

**Comment by tdaede** [comment]
Superseded by #341.

**Pull request closed without merging**